File: BACK-END/KuentecoChat/src/main/java/com/example/back_end/KuentecoChat/exception/ApiResponse.java
package com.example.back_end.KuentecoChat.exception;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class ApiResponse<T> {
    private boolean success;
    private String message;
    private T data;
    private LocalDateTime timestamp;
    private String path; // útil para trazabilidad

    public static <T> ApiResponse<T> ok(String message, T data, String path) {
        return ApiResponse.<T> builder().success(true).message(message).data(data).timestamp(LocalDateTime.now())
                .path(path).build();
    }

    public static ApiResponse<Void> error(String message, String path) {
        return ApiResponse.<Void> builder().success(false).message(message).timestamp(LocalDateTime.now()).path(path)
                .build();
    }
}

File: FRONT-END/lib/core/exceptions/api_response.dart
class ApiResponse<T> {
  final bool success;
  final String message;
  final T data;
  final String path;
  final DateTime timestamp;

  ApiResponse({
    required this.success,
    required this.message,
    required this.data,
    required this.path,
    required this.timestamp,
  });

  factory ApiResponse.fromJson(
      Map<String, dynamic> json,
      T Function(dynamic) fromDataJson,
      ) {
    return ApiResponse<T>(
      success: json['success'],
      message: json['message'],
      data: fromDataJson(json['data']),
      path: json['path'],
      timestamp: DateTime.parse(json['timestamp']),
    );
  }
}